### ISSUE TITLE
False negative in :java-static-field-call linter

### ISSUE THREAD

**Issue by dpassen** [clj-kondo/clj-kondo#2456]
- [X] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

v2024.11.14

**platform**

macOS Sequoia 15.1
Binary build clj-kondo from homebrew

**editor**

emacs 29.4 flycheck-clj-kondo

**problem**

A third party JAR contains a static field, which we have been incorrectly invoking. The `:java-static-field-call` fails to find it.

**repro**

(also available at https://github.com/dpassen/kondo-static-field-repro)

```clj
io.temporal/temporal-sdk {:mvn/version "1.24.1"}
```

repro.clj

```clj
(ns repro
  (:import
   (clojure.lang PersistentQueue)
   (io.temporal.workflow Workflow)
   (java.time.temporal ChronoField)))

(System/err)
System/err

(PersistentQueue/EMPTY)
PersistentQueue/EMPTY

(Workflow/DEFAULT_VERSION)
Workflow/DEFAULT_VERSION

(ChronoField/DAY_OF_MONTH)
ChronoField/DAY_OF_MONTH

```

**config**

No config file used.

**expected behavior**

clj-kondo should find all four static fields being invoked, but only finds three.

***expected***

```sh
$ clj-kondo --lint src/
src/kondo_static_field_repro/core.clj:7:1: error: Static fields should be referenced without parens unless they are intended as function calls
src/kondo_static_field_repro/core.clj:10:1: error: Static fields should be referenced without parens unless they are intended as function calls
src/kondo_static_field_repro/core.clj:13:1: error: Static fields should be referenced without parens unless they are intended as function calls
src/kondo_static_field_repro/core.clj:16:1: error: Static fields should be referenced without parens unless they are intended as function calls
linting took 7ms, errors: 4, warnings: 0
```

***actual***

```sh
$ clj-kondo --lint src/
src/kondo_static_field_repro/core.clj:7:1: error: Static fields should be referenced without parens unless they are intended as function calls
src/kondo_static_field_repro/core.clj:10:1: error: Static fields should be referenced without parens unless they are intended as function calls
src/kondo_static_field_repro/core.clj:16:1: error: Static fields should be referenced without parens unless they are intended as function calls
linting took 7ms, errors: 3, warnings: 0
```

**Comment by borkdude** [comment]
Thanks!

**Comment by borkdude** [comment]
OK, I looked into this.

A few things:

- To analyze `io.temporal.workflow Workflow` you need to first analyze your dependencies
-  clj-kondo doesn't analyze java stuff by default so you have to enable that

So if you first execute this, before linting your dependencies for warnings, I'll work:

```
clj-kondo --lint "$(clojure -Spath)" --config '{:analysis {:java-class-definitions true, :java-member-definitions true}}'  --dependencies
```
